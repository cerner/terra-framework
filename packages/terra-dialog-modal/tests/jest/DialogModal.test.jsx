import React from 'react';
import DialogModalExample from './DialogModalExample';

it('should shallow an open dialogModal', () => {
  const dialogModal = shallow(<DialogModalExample />);
  expect(dialogModal).toMatchSnapshot();
});

it('should mount an open dialogModal', () => {
  const dialogModal = mount(<DialogModalExample />);
  expect(dialogModal).toMatchSnapshot();
});

it('should shallow an open dialogModal 320 width', () => {
  const dialogModal = shallow(<DialogModalExample width="320" />);
  expect(dialogModal).toMatchSnapshot();
});

it('should mount an open dialogModal 320 width', () => {
  const dialogModal = mount(<DialogModalExample width="320" />);
  expect(dialogModal).toMatchSnapshot();
});

it('should shallow an open dialogModal 960 width', () => {
  const dialogModal = shallow(<DialogModalExample width="960" />);
  expect(dialogModal).toMatchSnapshot();
});

it('should mount an open dialogModal 960 width', () => {
  const dialogModal = mount(<DialogModalExample width="960" />);
  expect(dialogModal).toMatchSnapshot();
});

it('should shallow an open dialogModal 1280 width', () => {
  const dialogModal = shallow(<DialogModalExample width="1280" />);
  expect(dialogModal).toMatchSnapshot();
});

it('should mount an open dialogModal 1280 width', () => {
  const dialogModal = mount(<DialogModalExample width="1280" />);
  expect(dialogModal).toMatchSnapshot();
});

it('should shallow an open dialogModal 1600 width', () => {
  const dialogModal = shallow(<DialogModalExample width="1600" />);
  expect(dialogModal).toMatchSnapshot();
});

it('should mount an open dialogModal 1600 width', () => {
  const dialogModal = mount(<DialogModalExample width="1600" />);
  expect(dialogModal).toMatchSnapshot();
});

it('should mount an open dialogModal 1600 width', () => {
  const dialogModal = mount(<DialogModalExample width="1600" />);
  expect(dialogModal).toMatchSnapshot();
});

it('should set the rootSelector', () => {
  const dialogModal = mount(<DialogModalExample rootSelector="#myroot" />);
  expect(dialogModal).toMatchSnapshot();
});

it('correctly applies the theme context className', () => {
  jest.spyOn(React, 'useContext')
    .mockReturnValue({
      className: 'orion-fusion-theme',
    });
  const wrapper = mount(<DialogModalExample />);
  expect(wrapper).toMatchSnapshot();
});
