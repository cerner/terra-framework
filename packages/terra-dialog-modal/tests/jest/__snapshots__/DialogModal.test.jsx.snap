// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should mount an open dialogModal 1`] = `
<DialogModalExample
  isFocused={false}
>
  <div>
    <DialogModal
      ariaLabel="Dialog Modal Example"
      footer={
        <ActionFooter
          start="Footer Goes here"
        />
      }
      header={
        <ActionHeader
          level={1}
          onClose={[Function]}
          title="Action Header used here"
        />
      }
      isFocused={false}
      isOpen={false}
      onRequestClose={[Function]}
      releaseFocus={null}
      requestFocus={null}
      width="1120"
    />
    <Button
      isBlock={false}
      isCompact={false}
      isDisabled={false}
      isIconOnly={false}
      isReversed={false}
      onClick={[Function]}
      text="Trigger Dialog Modal"
      type="button"
      variant="neutral"
    >
      <button
        aria-disabled={false}
        className="button neutral"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        type="button"
      >
        <span
          className="button-label text-only"
        >
          <span
            className=""
          >
            Trigger Dialog Modal
          </span>
        </span>
      </button>
    </Button>
  </div>
</DialogModalExample>
`;

exports[`should mount an open dialogModal 320 width 1`] = `
<DialogModalExample
  isFocused={false}
  width="320"
>
  <div>
    <DialogModal
      ariaLabel="Dialog Modal Example"
      footer={
        <ActionFooter
          start="Footer Goes here"
        />
      }
      header={
        <ActionHeader
          level={1}
          onClose={[Function]}
          title="Action Header used here"
        />
      }
      isFocused={false}
      isOpen={false}
      onRequestClose={[Function]}
      releaseFocus={null}
      requestFocus={null}
      width="320"
    />
    <Button
      isBlock={false}
      isCompact={false}
      isDisabled={false}
      isIconOnly={false}
      isReversed={false}
      onClick={[Function]}
      text="Trigger Dialog Modal"
      type="button"
      variant="neutral"
    >
      <button
        aria-disabled={false}
        className="button neutral"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        type="button"
      >
        <span
          className="button-label text-only"
        >
          <span
            className=""
          >
            Trigger Dialog Modal
          </span>
        </span>
      </button>
    </Button>
  </div>
</DialogModalExample>
`;

exports[`should mount an open dialogModal 960 width 1`] = `
<DialogModalExample
  isFocused={false}
  width="960"
>
  <div>
    <DialogModal
      ariaLabel="Dialog Modal Example"
      footer={
        <ActionFooter
          start="Footer Goes here"
        />
      }
      header={
        <ActionHeader
          level={1}
          onClose={[Function]}
          title="Action Header used here"
        />
      }
      isFocused={false}
      isOpen={false}
      onRequestClose={[Function]}
      releaseFocus={null}
      requestFocus={null}
      width="960"
    />
    <Button
      isBlock={false}
      isCompact={false}
      isDisabled={false}
      isIconOnly={false}
      isReversed={false}
      onClick={[Function]}
      text="Trigger Dialog Modal"
      type="button"
      variant="neutral"
    >
      <button
        aria-disabled={false}
        className="button neutral"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        type="button"
      >
        <span
          className="button-label text-only"
        >
          <span
            className=""
          >
            Trigger Dialog Modal
          </span>
        </span>
      </button>
    </Button>
  </div>
</DialogModalExample>
`;

exports[`should mount an open dialogModal 1280 width 1`] = `
<DialogModalExample
  isFocused={false}
  width="1280"
>
  <div>
    <DialogModal
      ariaLabel="Dialog Modal Example"
      footer={
        <ActionFooter
          start="Footer Goes here"
        />
      }
      header={
        <ActionHeader
          level={1}
          onClose={[Function]}
          title="Action Header used here"
        />
      }
      isFocused={false}
      isOpen={false}
      onRequestClose={[Function]}
      releaseFocus={null}
      requestFocus={null}
      width="1280"
    />
    <Button
      isBlock={false}
      isCompact={false}
      isDisabled={false}
      isIconOnly={false}
      isReversed={false}
      onClick={[Function]}
      text="Trigger Dialog Modal"
      type="button"
      variant="neutral"
    >
      <button
        aria-disabled={false}
        className="button neutral"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        type="button"
      >
        <span
          className="button-label text-only"
        >
          <span
            className=""
          >
            Trigger Dialog Modal
          </span>
        </span>
      </button>
    </Button>
  </div>
</DialogModalExample>
`;

exports[`should mount an open dialogModal 1600 width 1`] = `
<DialogModalExample
  isFocused={false}
  width="1600"
>
  <div>
    <DialogModal
      ariaLabel="Dialog Modal Example"
      footer={
        <ActionFooter
          start="Footer Goes here"
        />
      }
      header={
        <ActionHeader
          level={1}
          onClose={[Function]}
          title="Action Header used here"
        />
      }
      isFocused={false}
      isOpen={false}
      onRequestClose={[Function]}
      releaseFocus={null}
      requestFocus={null}
      width="1600"
    />
    <Button
      isBlock={false}
      isCompact={false}
      isDisabled={false}
      isIconOnly={false}
      isReversed={false}
      onClick={[Function]}
      text="Trigger Dialog Modal"
      type="button"
      variant="neutral"
    >
      <button
        aria-disabled={false}
        className="button neutral"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        type="button"
      >
        <span
          className="button-label text-only"
        >
          <span
            className=""
          >
            Trigger Dialog Modal
          </span>
        </span>
      </button>
    </Button>
  </div>
</DialogModalExample>
`;

exports[`should mount an open dialogModal 1600 width 2`] = `
<DialogModalExample
  isFocused={false}
  width="1600"
>
  <div>
    <DialogModal
      ariaLabel="Dialog Modal Example"
      footer={
        <ActionFooter
          start="Footer Goes here"
        />
      }
      header={
        <ActionHeader
          level={1}
          onClose={[Function]}
          title="Action Header used here"
        />
      }
      isFocused={false}
      isOpen={false}
      onRequestClose={[Function]}
      releaseFocus={null}
      requestFocus={null}
      width="1600"
    />
    <Button
      isBlock={false}
      isCompact={false}
      isDisabled={false}
      isIconOnly={false}
      isReversed={false}
      onClick={[Function]}
      text="Trigger Dialog Modal"
      type="button"
      variant="neutral"
    >
      <button
        aria-disabled={false}
        className="button neutral"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        type="button"
      >
        <span
          className="button-label text-only"
        >
          <span
            className=""
          >
            Trigger Dialog Modal
          </span>
        </span>
      </button>
    </Button>
  </div>
</DialogModalExample>
`;

exports[`should mount an open dialogModal with focus released 1`] = `
<DialogModalExample
  isFocused={false}
  width="1600"
>
  <div>
    <DialogModal
      ariaLabel="Dialog Modal Example"
      footer={
        <ActionFooter
          start="Footer Goes here"
        />
      }
      header={
        <ActionHeader
          level={1}
          onClose={[Function]}
          title="Action Header used here"
        />
      }
      isFocused={false}
      isOpen={false}
      onRequestClose={[Function]}
      releaseFocus={null}
      requestFocus={null}
      width="1600"
    />
    <Button
      isBlock={false}
      isCompact={false}
      isDisabled={false}
      isIconOnly={false}
      isReversed={false}
      onClick={[Function]}
      text="Trigger Dialog Modal"
      type="button"
      variant="neutral"
    >
      <button
        aria-disabled={false}
        className="button neutral"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        type="button"
      >
        <span
          className="button-label text-only"
        >
          <span
            className=""
          >
            Trigger Dialog Modal
          </span>
        </span>
      </button>
    </Button>
  </div>
</DialogModalExample>
`;

exports[`should mount an open dialogModal with focus requested 1`] = `
<DialogModalExample
  isFocused={true}
  width="1600"
>
  <div>
    <DialogModal
      ariaLabel="Dialog Modal Example"
      footer={
        <ActionFooter
          start="Footer Goes here"
        />
      }
      header={
        <ActionHeader
          level={1}
          onClose={[Function]}
          title="Action Header used here"
        />
      }
      isFocused={true}
      isOpen={false}
      onRequestClose={[Function]}
      releaseFocus={null}
      requestFocus={null}
      width="1600"
    />
    <Button
      isBlock={false}
      isCompact={false}
      isDisabled={false}
      isIconOnly={false}
      isReversed={false}
      onClick={[Function]}
      text="Trigger Dialog Modal"
      type="button"
      variant="neutral"
    >
      <button
        aria-disabled={false}
        className="button neutral"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        type="button"
      >
        <span
          className="button-label text-only"
        >
          <span
            className=""
          >
            Trigger Dialog Modal
          </span>
        </span>
      </button>
    </Button>
  </div>
</DialogModalExample>
`;

exports[`should shallow an open dialogModal 1`] = `
<div>
  <DialogModal
    ariaLabel="Dialog Modal Example"
    footer={
      <ActionFooter
        start="Footer Goes here"
      />
    }
    header={
      <ActionHeader
        level={1}
        onClose={[Function]}
        title="Action Header used here"
      />
    }
    isFocused={false}
    isOpen={false}
    onRequestClose={[Function]}
    releaseFocus={null}
    requestFocus={null}
    width="1120"
  >
    <p>
      Lorem ipsum dolor sit amet, consectetur adipiscing elit. 
      Maecenas molestie in lorem vel facilisis. Quisque ac enim nec lectus malesuada faucibus.
      Integer congue feugiat ultricies.
       Nunc non mauris sed tellus cursus vestibulum nec quis ipsum.
      Vivamus ornare magna justo, et volutpat tortor congue ut. Nulla facilisi.
       Cras in venenatis turpis. Nullam id odio justo. Etiam vehicula lectus vel purus consectetur cursus id sit amet diam.
      Donec facilisis dui non orci hendrerit pharetra. Suspendisse blandit dictum turpis, in consectetur ipsum hendrerit eget.
      Nam vehicula, arcu vitae egestas porttitor,
      turpis nisi pulvinar neque, ut lacinia urna purus sit amet elit.
    </p>
  </DialogModal>
  <Button
    isBlock={false}
    isCompact={false}
    isDisabled={false}
    isIconOnly={false}
    isReversed={false}
    onClick={[Function]}
    text="Trigger Dialog Modal"
    type="button"
    variant="neutral"
  />
</div>
`;

exports[`should shallow an open dialogModal 320 width 1`] = `
<div>
  <DialogModal
    ariaLabel="Dialog Modal Example"
    footer={
      <ActionFooter
        start="Footer Goes here"
      />
    }
    header={
      <ActionHeader
        level={1}
        onClose={[Function]}
        title="Action Header used here"
      />
    }
    isFocused={false}
    isOpen={false}
    onRequestClose={[Function]}
    releaseFocus={null}
    requestFocus={null}
    width="320"
  >
    <p>
      Lorem ipsum dolor sit amet, consectetur adipiscing elit. 
      Maecenas molestie in lorem vel facilisis. Quisque ac enim nec lectus malesuada faucibus.
      Integer congue feugiat ultricies.
       Nunc non mauris sed tellus cursus vestibulum nec quis ipsum.
      Vivamus ornare magna justo, et volutpat tortor congue ut. Nulla facilisi.
       Cras in venenatis turpis. Nullam id odio justo. Etiam vehicula lectus vel purus consectetur cursus id sit amet diam.
      Donec facilisis dui non orci hendrerit pharetra. Suspendisse blandit dictum turpis, in consectetur ipsum hendrerit eget.
      Nam vehicula, arcu vitae egestas porttitor,
      turpis nisi pulvinar neque, ut lacinia urna purus sit amet elit.
    </p>
  </DialogModal>
  <Button
    isBlock={false}
    isCompact={false}
    isDisabled={false}
    isIconOnly={false}
    isReversed={false}
    onClick={[Function]}
    text="Trigger Dialog Modal"
    type="button"
    variant="neutral"
  />
</div>
`;

exports[`should shallow an open dialogModal 960 width 1`] = `
<div>
  <DialogModal
    ariaLabel="Dialog Modal Example"
    footer={
      <ActionFooter
        start="Footer Goes here"
      />
    }
    header={
      <ActionHeader
        level={1}
        onClose={[Function]}
        title="Action Header used here"
      />
    }
    isFocused={false}
    isOpen={false}
    onRequestClose={[Function]}
    releaseFocus={null}
    requestFocus={null}
    width="960"
  >
    <p>
      Lorem ipsum dolor sit amet, consectetur adipiscing elit. 
      Maecenas molestie in lorem vel facilisis. Quisque ac enim nec lectus malesuada faucibus.
      Integer congue feugiat ultricies.
       Nunc non mauris sed tellus cursus vestibulum nec quis ipsum.
      Vivamus ornare magna justo, et volutpat tortor congue ut. Nulla facilisi.
       Cras in venenatis turpis. Nullam id odio justo. Etiam vehicula lectus vel purus consectetur cursus id sit amet diam.
      Donec facilisis dui non orci hendrerit pharetra. Suspendisse blandit dictum turpis, in consectetur ipsum hendrerit eget.
      Nam vehicula, arcu vitae egestas porttitor,
      turpis nisi pulvinar neque, ut lacinia urna purus sit amet elit.
    </p>
  </DialogModal>
  <Button
    isBlock={false}
    isCompact={false}
    isDisabled={false}
    isIconOnly={false}
    isReversed={false}
    onClick={[Function]}
    text="Trigger Dialog Modal"
    type="button"
    variant="neutral"
  />
</div>
`;

exports[`should shallow an open dialogModal 1280 width 1`] = `
<div>
  <DialogModal
    ariaLabel="Dialog Modal Example"
    footer={
      <ActionFooter
        start="Footer Goes here"
      />
    }
    header={
      <ActionHeader
        level={1}
        onClose={[Function]}
        title="Action Header used here"
      />
    }
    isFocused={false}
    isOpen={false}
    onRequestClose={[Function]}
    releaseFocus={null}
    requestFocus={null}
    width="1280"
  >
    <p>
      Lorem ipsum dolor sit amet, consectetur adipiscing elit. 
      Maecenas molestie in lorem vel facilisis. Quisque ac enim nec lectus malesuada faucibus.
      Integer congue feugiat ultricies.
       Nunc non mauris sed tellus cursus vestibulum nec quis ipsum.
      Vivamus ornare magna justo, et volutpat tortor congue ut. Nulla facilisi.
       Cras in venenatis turpis. Nullam id odio justo. Etiam vehicula lectus vel purus consectetur cursus id sit amet diam.
      Donec facilisis dui non orci hendrerit pharetra. Suspendisse blandit dictum turpis, in consectetur ipsum hendrerit eget.
      Nam vehicula, arcu vitae egestas porttitor,
      turpis nisi pulvinar neque, ut lacinia urna purus sit amet elit.
    </p>
  </DialogModal>
  <Button
    isBlock={false}
    isCompact={false}
    isDisabled={false}
    isIconOnly={false}
    isReversed={false}
    onClick={[Function]}
    text="Trigger Dialog Modal"
    type="button"
    variant="neutral"
  />
</div>
`;

exports[`should shallow an open dialogModal 1600 width 1`] = `
<div>
  <DialogModal
    ariaLabel="Dialog Modal Example"
    footer={
      <ActionFooter
        start="Footer Goes here"
      />
    }
    header={
      <ActionHeader
        level={1}
        onClose={[Function]}
        title="Action Header used here"
      />
    }
    isFocused={false}
    isOpen={false}
    onRequestClose={[Function]}
    releaseFocus={null}
    requestFocus={null}
    width="1600"
  >
    <p>
      Lorem ipsum dolor sit amet, consectetur adipiscing elit. 
      Maecenas molestie in lorem vel facilisis. Quisque ac enim nec lectus malesuada faucibus.
      Integer congue feugiat ultricies.
       Nunc non mauris sed tellus cursus vestibulum nec quis ipsum.
      Vivamus ornare magna justo, et volutpat tortor congue ut. Nulla facilisi.
       Cras in venenatis turpis. Nullam id odio justo. Etiam vehicula lectus vel purus consectetur cursus id sit amet diam.
      Donec facilisis dui non orci hendrerit pharetra. Suspendisse blandit dictum turpis, in consectetur ipsum hendrerit eget.
      Nam vehicula, arcu vitae egestas porttitor,
      turpis nisi pulvinar neque, ut lacinia urna purus sit amet elit.
    </p>
  </DialogModal>
  <Button
    isBlock={false}
    isCompact={false}
    isDisabled={false}
    isIconOnly={false}
    isReversed={false}
    onClick={[Function]}
    text="Trigger Dialog Modal"
    type="button"
    variant="neutral"
  />
</div>
`;
